src/components/IdentifierEditCard.vue

```diff
@@ -86,16 +80,25 @@ export default defineComponent({
             default: ''
         },
         value: {
-            type: String,
+            type: String as PropType<IdentifierTypeType>,
             default: ''
         },
         description: {
             type: String,
             default: ''
         }
     },
-    setup () {
+    setup (props) {
+        // validating of value depends on type
+        const valueValidators: Record<IdentifierTypeType, (val: unknown) => true | string > = {
+            doi: makeFieldValidator('/definitions/identifier/anyOf/0/properties/value'),
+            url: makeFieldValidator('/definitions/identifier/anyOf/1/properties/value'),
+            swh: makeFieldValidator('/definitions/identifier/anyOf/1/properties/value'),
```

Comment:
Would have expected 2 here, like so:

```suggestion
            swh: makeFieldValidator('/definitions/identifier/anyOf/2/properties/value'),
```
or maybe a more direct route, `#/definitions/swh-identifier`

Reply:
The subschema for `type=other` needs this long version so I made each type use same subschema structure.
Fixed it